Doc/library/shelve.rst

```diff
@@ -17,7 +17,7 @@ This includes most class instances, recursive data types, and objects containing
 lots of shared  sub-objects.  The keys are ordinary strings.
 
 
-.. function:: open(filename, flag='c', protocol=None, writeback=False)
+.. function:: open(filename, flag='c', protocol=None, writeback=False, custom_dumps=dumps, custom_loads=loads)
```

Comment:
I'd either go with `*, serializer=pickle.dumps, deserializer=pickle.loads)` or `dumps/loads` (dropping the custom prefixes).

Reply:
In the event that someone only passes the proper `serializer` and drops the `deserializer`, having it to default `pickle.loads` seems wrong although I don't have any better options either. I guess the best we can offer is to make these `None` by default, and handle the default logic in the code where we would ensure both are untouched and then inherit pickle as the default method.

Reply:
The signature of the function has been changed to 
```python
open(filename, flag='c', protocol=None, writeback=False, serializer=None, deserializer=None)
```